src/test/ui/issues/issue-69455.stderr

```diff
@@ -0,0 +1,17 @@
+error[E0284]: type annotations needed
+  --> $DIR/issue-69455.rs:29:26
+   |
+LL |     type Output;
+   |     ------------ `<Self as Test<Rhs>>::Output` defined here
+...
+LL |     println!("{}", 23u64.test(xs.iter().sum()));
+   |                    ------^^^^-----------------
+   |                    |     |
+   |                    |     cannot infer type for type `u64`
+   |                    this method call resolves to `<Self as Test<Rhs>>::Output`
```

Comment:
It would be amazing if we actually suggested `<u64 as Test<u64>>::test(23u64, xs.iter().sum())`, but that is beyond the scope of this PR.